test/crawler/keyword/keywords_test.exs

```diff
@@ -5,15 +5,77 @@ defmodule Crawler.Keyword.KeywordsTest do
   alias Crawler.Keyword.Schemas.Keyword
   alias CrawlerWorker.Keyword.CrawlerWorker
 
-  describe "list_keywords/1" do
+  describe "list_keywords_by_filter_params/1" do
     test "given a valid user id, returns user's keyword list" do
       %{id: user_id} = insert(:user)
       insert(:keyword, user_id: user_id)
 
       %{id: expected_user_id} = insert(:user)
       expected_keyword = insert(:keyword, user_id: expected_user_id)
 
-      assert Keywords.list_keywords(expected_user_id) == [expected_keyword]
+      assert Keywords.list_keywords_by_filter_params(expected_user_id) == [expected_keyword]
+    end
+  end
+
+  describe "list_keywords_by_filter_params/2" do
+    test "given a valid user id and valid keyword name, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      expected_keyword = insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => expected_keyword.name
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given a part of valid keyword, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => "hon"
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given an invalid keyword, returns empty keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert Keywords.list_keywords_by_filter_params(expected_user_id, %{
+               "keyword" => "something"
+             }) == []
+    end
+
+    test "given a nil keyword, returns keyword list" do
```

Comment:
should we add a test for other filter param

```
context 'given the filter is not `keyword``
  it 'doesn't filter....`
```

Reply:
Added in [6bd2542](https://github.com/vnntsu/elixir-ic/pull/44/commits/6bd254207c503d8f6476d6ca4c09ee1f952d6dbc)